llvm/include/llvm/TargetParser/AArch64TargetParser.h

```diff
@@ -161,14 +161,10 @@ struct CpuInfo {
   StringRef Name; // Name, as written for -mcpu.
   const ArchInfo &Arch;
   AArch64::ExtensionBitset
-      DefaultExtensions; // Default extensions for this CPU. These will be
-                         // ORd with the architecture defaults.
+      DefaultExtensions; // Default extensions for this CPU.
 
   AArch64::ExtensionBitset getImpliedExtensions() const {
-    AArch64::ExtensionBitset ImpliedExts;
-    ImpliedExts |= DefaultExtensions;
-    ImpliedExts |= Arch.DefaultExts;
-    return ImpliedExts;
+    return DefaultExtensions;
```

Comment:
IIUC, the idea here is that we don't want to add the architecture's default extensions to the CPU, because they are not mandatory, and there's no way to disable them for a CPU that chooses not to implement. This is why all the CPUs have had their list of features expanded.

Reply:
Yes, and that idea is the problem: automatically adding default extensions kinda works for extensions that the compiler doesn't generate automatically where their use is driven by intrinsics, but as soon as you have something like SVE in the list you get incompatible codegen left and right.

Reply:
This choice also makes `-march=` meaningless, so @ahmedbougacha and I are considering making that an error at least on Darwin platforms.

Reply:
Not sure I follow... Do you mean `-march=` has no effect if you specify `-mcpu`? That should probably be at least a warning.

Reply:
I mean that code built with `-march=armv9.2` may use SVE instructions because `FeatureSVE` is in the default list for `v9.0`, but this will crash on an M4 iPad.

Reply:
for example: https://clang.godbolt.org/z/4jW8nqK41